File: Chess/DataTypes/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataTypes
{
    public class Player
    {
        public string Name { get; set; } = "";
        public int Money { get; set; } = 0;
        public bool IsCurrent = false;
        private Action<Player> refresh;
        public Card[] Hand;

        public Player(string name, int money, Action<Player> refresh)
        {
            Name = name;
            Money = money;
            this.refresh = refresh;
        }

        public void Refresh()
        {
            refresh(this);
        }

    }
}

File: Chess/NodeChessServer/DataTypes/Player.js
'use strict';
class Player {
    constructor(name, money, refresh, lose, win, step, info) {
        this.name = name;
        this.money = money;
        this.isCurrent = false;
        this.refresh = () => { refresh(this) };
        this.step = (num) => { step(num); }
        this.lose = lose;
        this.win = win;
        this.info = info;
        this.hand = [];
    }

    Refresh() {
        this.refresh(this);
    }
}

module.exports = Player;